ggshield/core/scan/commit.py

```diff
@@ -73,6 +76,39 @@ def parser(commit: "Commit") -> Iterable[Scannable]:
 
         return Commit(sha, parser, info)
 
+    @staticmethod
+    def from_merge(
+        exclusion_regexes: Optional[Set[Pattern[str]]] = None,
+        merge_branch: str = "MERGE_HEAD",
+        cwd: Optional[Path] = None,
+    ) -> "Commit":
+
+        diff_files = get_diff_files(cwd=cwd)
+
+        shas_in_merge_branch = dict(
+            get_file_sha_in_ref(merge_branch, diff_files, cwd=cwd)
+        )
+        shas_in_head = dict(get_file_sha_in_ref("HEAD", diff_files, cwd=cwd))
+
+        files_to_scan = set()
+        for path, sha in get_file_sha_stage(diff_files, cwd=cwd):
+            # The file is either new or different from both HEAD and MERGE_HEAD
+            if sha not in {shas_in_head.get(path), shas_in_merge_branch.get(path)}:
+                files_to_scan.add(path)
+
+        def parser_merge(commit: "Commit") -> Iterable[Scannable]:
+            patch = git(
+                ["diff", "--staged"] + PATCH_COMMON_ARGS + list(files_to_scan), cwd=cwd
```

Comment:
nit, can be simplified to:

```suggestion
                ["diff", "--staged", *PATCH_COMMON_ARGS, *files_to_scan], cwd=cwd
```

This avoids creating an intermediate list for `files_to_scan`.